File: chrome/android/java/src/org/chromium/chrome/browser/BrowserVersion.java
// Copyright 2014 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser;

/**
 * TODO(nileshagrawal): Rename this class to something more appropriate.
 * Provides a way for java code to determine whether Chrome was built as an official build.
 */
public class BrowserVersion {
    /**
     * Check if the browser was built as an "official" build.
     *
     * This function depends on the native library being loaded; calling it before then will crash.
     */
    public static boolean isOfficialBuild() {
        return nativeIsOfficialBuild();
    }

    private static native boolean nativeIsOfficialBuild();
}

File: chrome/browser/android/browser_version.cc
// Copyright 2014 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/browser/android/browser_version.h"

#include "chrome/common/chrome_version_info.h"
#include "jni/BrowserVersion_jni.h"

static jboolean IsOfficialBuild(JNIEnv* env, jclass /* clazz */) {
  return chrome::VersionInfo().IsOfficialBuild();
}

bool RegisterBrowserVersion(JNIEnv* env) {
  return RegisterNativesImpl(env);
}
